nixos/modules/services/monitoring/tuptime.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ...}:
+
+with lib;
+
+let
+
+  cfg = config.services.tuptime;
+
+in {
+
+  options.services.tuptime = {
+
+    enable = mkEnableOption "the total uptime service";
+
+    timer = {
+      enable = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to regularly log uptime to detect bad shutdowns.";
+      };
+
+      period = mkOption {
+        type = types.str;
+        default = "*:0/5";
+        description = "systemd calendar event";
+      };
+    };
+  };
+
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.tuptime ];
+
+    users.users.tuptime = {
+      description = "tuptime database owner";
+      home = "/var/lib/tuptime";
+      createHome = true;
+    };
+
+    systemd = {
+      services = {
+
+        tuptime = {
+          description = "the total uptime service";
+          documentation = [ "man:tuptime(1)" ];
+          wants = [ "time-sync.target" ];
+          wantedBy = [ "multi-user.target" ];
+          preStart = "chmod 755 /var/lib/tuptime";
```

Comment:
Perhaps this should be a `mkdir -p -m 755` in a wrapper for the executable?

Reply:
You're looking for `serviceConfig.StateDirectory`.

Reply:
you're a godsend!